src/libsync/abstractnetworkjob.h

```diff
@@ -24,13 +24,71 @@
 #include <QDateTime>
 #include <QTimer>
 #include "accountfwd.h"
+#include "common/asserts.h"
 
 class QUrl;
 
 namespace OCC {
 
 class AbstractSslErrorHandler;
 
+
+/**
+ * A Result of type T, or an Error that contains a code and a string
+ **/
+template <typename T>
+class Result
+{
+    struct Error
+    {
+        QString string;
+        int code;
+    };
+    union {
+        T _result;
+        Error _error;
+    };
+    bool _isError;
+
+public:
+    Result(T value)
+        : _result(std::move(value))
+        , _isError(false){};
+    Result(int code, QString str)
+        : _error({ std::move(str), code })
+        , _isError(true)
+    {
+    }
+    ~Result()
+    {
+        if (_isError)
+            _error.~Error();
+        else
+            _result.~T();
+    }
+    explicit operator bool() const { return !_isError; }
+    const T &operator*() const &
+    {
+        ASSERT(!_isError);
+        return _result;
+    }
+    T operator*() &&
+    {
+        ASSERT(!_isError);
+        return std::move(_result);
+    }
+    QString errorMessage() const
+    {
+        ASSERT(_isError);
+        return _error.string;
+    }
+    int errorCode() const
+    {
+        return _isError ? _error.code : 0;
+    }
+};
```

Comment:
Right...  I just needed it, and did not want to implement too much, knowing how much bikeshed there can be for such utility class.

It's indeed based on rust's `Result`, and LLVM's `Expected` class. (there is also a std::expected proposal, but it is just a source of bikeshedding :-))
I liked the name Result better than Expected, so i named it result.  But it can be easilly changed as there are only a few use of it right now. (altough i would plan to use it everywhere in the future)